source/extensions/filters/common/expr/custom_cel/extended_request/extended_request_cel_vocabulary.h

```diff
@@ -0,0 +1,76 @@
+#pragma once
+
+#include "envoy/extensions/expr/custom_cel_vocabulary/extended_request/v3/config.pb.h"
+#include "envoy/protobuf/message_validator.h"
+
+#include "source/common/protobuf/protobuf.h"
+#include "source/extensions/filters/common/expr/context.h"
+#include "source/extensions/filters/common/expr/custom_cel/custom_cel_vocabulary.h"
+#include "source/extensions/filters/common/expr/custom_cel/extended_request/custom_cel_functions.h"
+
+#include "absl/strings/string_view.h"
+#include "eval/public/activation.h"
+#include "eval/public/cel_function_registry.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace Filters {
+namespace Common {
+namespace Expr {
+namespace CustomCel {
+namespace ExtendedRequest {
+
+// Implementation of the CustomCelVocabulary interface.
+
+// activation value producer names
+constexpr absl::string_view ExtendedRequest = Request;
+
+// function names
+constexpr absl::string_view LazyFuncNameCookie = "cookie";
```

Comment:
The API is about query params, but here we are talking about cookies, this seems a bit confusing.

Reply:
The api was meant to include any configuration for the `request` attribute.  
It can include zero to many configuration parameters.  

There is configuration for the `query` field in `request` but no configuration is required for `cookie()` or `cookieValue()`

I can clarify the comments if it's not clear.

Reply:
Yeah, I think it should be fully described in the API docs what the new vocab provided is at the syntax level.